src/conn/conn_oligarch.c

```diff
@@ -93,6 +88,63 @@ __disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t checkpoint_id)
     /* Put our new config in */
     WT_ERR(__wt_metadata_insert(internal_session, metadata_key, cfg_ret));
 
+    /*
+     * Part 2: Get the metadata for other tables from the shared metadata table.
+     */
+
+    /* We need a separate internal session to pick up the new checkpoint. */
+    WT_ERR(__wt_open_internal_session(
+      conn, "checkpoint-pick-up-shared", false, 0, 0, &shared_metadata_session));
+
+    /* Scan the metadata table. */
+    cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_open_cursor);
+    cfg[1] = "checkpoint=" WT_CHECKPOINT ",checkpoint_use_history=false";
```

Comment:
Should we open the exact checkpoint instead of the latest checkpoint? There may be a new checkpoint done currently.

Reply:
My concern is not the checkpoint goes backwards. For example, we received the information that there is a new checkpoint 2 and we try to open it. However, we actually open checkpoint 3 instead of 2. Is there any problem in this case?

Reply:
I don't think that would happen. We request a table based on a specific checkpoint ID, which would be 2 in this case.

I actually just realized that this function is called with the checkpoint lock (see the lock assert at the top of the function), so we wouldn't even get another checkpoint in the meantime. So we would be still reading checkpoint 2 in this case.

Reply:
Thanks for the explaination.

Reply:
Yes, I will add a comment.